Feature/1841 ephemerides recenter floats

JulianHammerl opened the pull request:
* **Tickets addressed:** ema-gnc-1841
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
<!-- What approach was taken to satisfy the ticket being addressed? What should reviewers be aware of? How are your
commits organized? -->
The ephemeridesRecenter module is updated to use floats instead of doubles. The module is also refactored where needed.

## Verification
<!-- How were the changes validated? Were any automated tests added, updated, removed, or re-baselined? If you didn't 
add or update any tests justify this choice. -->
All tests pass.

## Documentation
<!-- What documentation was invalidated by these changes? Which artifacts should reviewers check for accuracy and 
completeness? -->
N/A

## Future work
<!-- What next steps can we anticipate from here, if any? -->
The doubles of translational states will be converted at a later point.

JulianHammerl committed:
Replace stdexcept with custom macro in algorithm

JulianHammerl committed:
Update includes

JulianHammerl committed:
Update license

JulianHammerl committed:
Formatting changes

codyallard28 commented on algorithms/ephemeridesRecenter/_tests/test_ephemeridesRecenter.py:
```diff
@@ -151,11 +151,11 @@ def mars_central_body():
     np.testing.assert_array_almost_equal(sun_v_recentered,
                                   np.array(sunInputPayload.v_BdyZero_N) - np.array(marsInputPayload.v_BdyZero_N),
-                                         15,
+                                         6,
```
If we are sticking with doubles, shouldn't these stay at double precision?

JulianHammerl replied:
Agreed, fixed. I had already changed this before we went back to keep translational states as doubles for now, so missed to undo this.

codyallard28 commented on algorithms/ephemeridesRecenter/ephemeridesRecenter.h:
```diff
@@ -39,8 +39,8 @@ class BodyEphemeris {
-    void updateState(uint64_t callTime) override;
-    void reset(uint64_t callTime) override;
+    void updateState(const uint64_t callTime) override;
+    void reset(const uint64_t callTime) override;
```
Same point about sonar lint/qube on the previous PR.

JulianHammerl replied:
Fixed.

codyallard28 reviewed (approved):
I just want to make sure the test reflects the right precision that we are expecting.

JulianHammerl committed:
Convert to FP32

JulianHammerl committed:
Use const where possible

JulianHammerl committed:
Remove unused messaging

JulianHammerl committed:
Import ephemeridesRecenter module

JulianHammerl committed:
Change Basilisk to xmera in unit tests and rst docs

JulianHammerl committed:
Use double for translational states and time for now

This will be changed later. But due to the large values of positions
and time, this requires some extra attention for float conversions.

JulianHammerl committed:
Change module definitions to get it to build
